File: email.verification/apikey/nodejs/email_apikey.js
const https = require('https');
const queryString = require('querystring');
const crypto = require('crypto');

const username = 'username';
const apiKey = 'api_key';
const secretKey = 'secret_key';
const url = 'https://whoisxmlapi.com/whoisserver/EmailVerifyService?';
const emailAddress = 'support@whoisxmlapi.com';

getWhois(username, apiKey, secretKey, emailAddress);

function getWhois(username, apiKey, secretKey, emailAddress)
{
    timestamp = (new Date).getTime();
    digest = generateDigest(username, timestamp, apiKey, secretKey);
    var requestString = buildRequest(username, timestamp, digest, emailAddress);
    https.get(url + requestString, function (res) {
        const statusCode = res.statusCode;

        if (statusCode !== 200) {
            console.log('Request failed: '
                + statusCode
            );
        }

        var rawData = '';

        res.on('data', function(chunk) {
            rawData += chunk;
        });

        res.on('end', function () {
            console.log(rawData);
        })
    }).on('error', function(e) {
        console.log("Error: " + e.message);
    });

}

function generateDigest(username, timestamp, apiKey, secretKey) {
    var data = username + timestamp + apiKey;
    var hmac = crypto.createHmac('md5', secretKey);
    hmac.update(data);
    return hmac.digest('hex');
}

function buildRequest(username, timestamp, digest, emailAddress) {
    var data = {
        u: username,
        t: timestamp
    };

    var dataJson = JSON.stringify(data);
    var dataBase64 = Buffer.from(dataJson).toString('base64');

    var request = {
        requestObject: dataBase64,
        digest: digest,
        emailAddress: emailAddress,
        validateDNS:true,
        validateSMTP:true,
        checkCatchAll:true,
        checkFree:true,
        checkDisposable:true,
        outputFormat:'json'
    };

    return queryString.stringify(request);
}

File: emailApi/apikey/perl/email_apikey.pl
use strict;
use warnings;
use Data::Dumper;
use LWP::Simple;                            # From CPAN
use JSON qw( decode_json encode_json );     # From CPAN
use Time::HiRes qw( time );                 # From CPAN
use Digest::HMAC_MD5 qw( hmac_md5_hex );    # From CPAN
use URI::Escape;                            # From CPAN
use MIME::Base64 qw( encode_base64 );

my $username = 'username';
my $apiKey = 'api_key';
my $secret = 'secret_key';

my $emailAddress = 'support@whoisxmlapi.com';
my $url = 'https://whoisxmlapi.com/whoisserver/EmailVerifyService?';

my $timestamp = int((time * 1000 + 0.5));
my $digest = generateDigest($username, $timestamp, $apiKey, $secret);

my $requstString = buildRequest(
    $username,
    $timestamp,
    $digest,
    $emailAddress
);
my $response = get($url . $requstString);
if (index($response, 'Request timeout')){
    $timestamp = int((time * 1000 + 0.5));
    $digest = generateDigest($username, $timestamp, $apiKey, $secret);
    $requstString = buildRequest(
        $username,
        $timestamp,
        $digest,
        $emailAddress
    );
    $response = get($url . $requstString);
}
printResponse($response);


sub generateDigest{
    my ($username, $timestamp, $apiKey, $secret) = @_;

    my $digest = $username . $timestamp . $apiKey;
    my $hash = hmac_md5_hex($digest, $secret);
    return uri_escape($hash);
}

sub buildRequest{
    my ($username, $timestamp, $digest, $emailAddress) = @_;
    my $requestString = 'requestObject=';
    my %request =(
        'u' => $username,
        't' => $timestamp
    );
    my $requestJson = encode_json(\%request);
    my $requestBase64 = encode_base64($requestJson);
    $requestString .= uri_escape($requestBase64);
    $requestString .= "&emailAddress=";
    $requestString .= $emailAddress;
    $requestString .= '&digest=';
    $requestString .= $digest;
    $requestString .= "&validateDNS=true&validateSMTP=true";
    $requestString .= "&checkCatchAll=true&checkFree=true&checkDisposable=true";
    $requestString .= '&outputFormat=json';
    return $requestString;
}

sub printResponse{
    my ($response) = @_;
    print $response;
    print "\n---------------------------------------\n";
}